sdk/typescript/src/eth-light-client/eth-light-client.ts

```diff
@@ -0,0 +1,305 @@
+// Copyright (c) dWallet Labs, Ltd.
+// SPDX-License-Identifier: BSD-3-Clause-Clear
+
+import { get_current_period, get_initial_state_bcs } from '@dwallet-network/eth-light-client-wasm';
+import { ethers } from 'ethers';
+
+import { bcs } from '../bcs/index.js';
+import { TransactionBlock } from '../builder/index.js';
+import type { DWalletClient } from '../client/index.js';
+import type { Keypair } from '../cryptography/index.js';
+import {
+	getBeaconBlockData,
+	getBootstrapData,
+	getFinalityUpdate,
+	getOptimisticUpdate,
+	getProof,
+	getUpdates,
+} from './rpc.js';
+import { getEthereumStateById, getLatestEthereumStateById, stringToArrayU8Bcs } from './utils.js';
+
+const packageId = '0x3';
+const ethDWalletModuleName = 'eth_dwallet';
+const ethereumStateModuleName = 'ethereum_state';
+
+/**
+ * Connects a dWallet to be controlled by an Ethereum smart contract.
+ *
+ * This function links a dWallet within the dWallet blockchain environment to an Ethereum smart contract.
+ * By creating an Ethereum dWallet capability, it allows the dWallet to interact with Ethereum transactions
+ * and be managed through the specified smart contract.
+ *
+ * **Arguments**
+ * @param {string} dwalletCapId - The ObjectID of the dWallet capability.
+ * @param {string} latestEthereumStateId - The ObjectID of the latest Ethereum state.
+ * @param {Keypair} keypair - The keypair used to sign the transaction.
+ * @param {DWalletClient} client - The dWallet client instance.
+ */
+export const createEthereumDWallet = async (
+	dwalletCapId: string,
+	latestEthereumStateId: string,
+	keypair: Keypair,
+	client: DWalletClient,
+) => {
+	const tx = new TransactionBlock();
+	tx.moveCall({
+		target: `${packageId}::${ethDWalletModuleName}::create_eth_dwallet_cap`,
+		arguments: [tx.object(dwalletCapId), tx.object(latestEthereumStateId)],
+	});
+	tx.setGasBudget(200000000);
+
+	let result = await client.signAndExecuteTransactionBlock({
+		signer: keypair,
+		transactionBlock: tx,
+		options: { showEffects: true },
+	});
+
+	if (result.effects?.created?.at(0)?.reference.objectId === undefined) {
+		console.log(result.effects);
```

Comment:
you are logging the effects and then returning the undefined object id? Seems like a weird behavior, maybe throw a descriptive error?

Reply:
this is a mistake